src/Federation/Repository/DeliveryQueueItem.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Federation\Repository;
+
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Federation\Collection;
+use Friendica\Federation\Entity;
+use Friendica\Federation\Factory;
+use Friendica\Util\DateTimeFormat;
+use Psr\Log\LoggerInterface;
+
+final class DeliveryQueueItem extends \Friendica\BaseRepository
+{
+	static $table_name = 'delivery-queue';
+
+	public function __construct(Database $database, LoggerInterface $logger, Factory\DeliveryQueueItem $factory)
+	{
+		parent::__construct($database, $logger, $factory);
+	}
+
+	public function selectByServerId(int $gsid, int $maxFailedCount): Collection\DeliveryQueueItems
+	{
+		$rows = $this->db->selectToArray(
```

Comment:
When you use `$this->db->select(...` and `$this->db->fetch(...` you reduce memory usage and increase the speed since we don't have to iterate twice through the entries.

Reply:
Done.